docs/rackhd/tags.rst

```diff
@@ -0,0 +1,147 @@
+Workflow Tag Support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Tag API provides functionality to automatically categorize nodes into 
+groups based on data present in a node's catalogs or by manually assigning 
+a tag to a node. When done automatically, tag matching is done using a
+series of rules. If all rules of a given tag match the latest version of 
+a node's catalog set, then that tag will be assigned to the node.  A node
+may be assigned many tags, both automatically through rules matching or 
+manually by the user.
+
+Upon discovering a node, the tag will be assigned based on all existing tag
+definitions in the system. Tags for all nodes will be re-generated whenever a
+tag definition is added.  Tags that are currently assigned to a node are not
+automatically removed from nodes when the rules backing a tag are deleted.
+
+
+**Example**
+
+
+With a node that has the following catalog fields:
+
+.. code-block:: javascript
+
+    {
+      "source": "dmi",
+      "data": {
+          "Base Board Information": {
+              "Manufacturer": "Intel Corporation"
+          }
+      },
+      "memory": {
+          "total": "32946864kB"
+          "free": "31682528kB"
+      }
+      /* ... */
+    }
+
+We could match against these fields with this tag definition:
+
+.. code-block:: javascript
+
+    {
+      "name": "Intel 32GB RAM",
+      "rules": [
+        {
+          "path": "dmi.Base Board Information.Manufacturer",
+          "contains": "Intel"
+        },
+        {
+          "path": "dmi.memory.total",
+          "equals": "32946864kB"
+        }
+      ]
+    }
+
+In both cases, the "path" string starts with "dmi" to signify that the rule
+should apply to the catalog with a "source" value of "dmi".
+
+This example makes use of the "contains" and "equals" rules. See the table at
+the bottom of this document for a list of additional validation rules that can
```

Comment:
I haven't see a table in the bottom of this doc that contains all the rules, is this one (http://rackhd.readthedocs.org/en/latest/rackhd/index.html#sku-json-format) ? I think adding a link here will be better.